indexing-service/src/main/java/io/druid/indexing/overlord/TaskQueue.java

```diff
@@ -79,12 +79,12 @@
   private final Condition managementMayBeNecessary = giant.newCondition();
   private final ExecutorService managerExec = Executors.newSingleThreadExecutor(
       new ThreadFactoryBuilder()
-          .setDaemon(false)
+          .setDaemon(false) // Don't use Execs because of this
```

Comment:
I'm not sure a comment is warranted here. Instead we should add java docs to execs clarifying that it's always a daemon thread, so someone is not tempted to replace this code.